reference-implementation/lib/utils.js

```diff
@@ -19,23 +19,56 @@ exports.tryURLParse = (string, baseURL) => {
 };
 
 exports.tryURLLikeSpecifierParse = (specifier, baseURL) => {
+  if (specifier === '') {
+    console.warn(`Invalid empty string specifier.`);
+    return { type: 'invalid' };
```

Comment:
Maybe it'd be better to communicate the warning as a field in the return type, which can then be turned into a thrown TypeError, or console.warn()ed, by the caller? (Related to above comment about tests that set console.warn to a no-op.)